src/onig-string.cc

```diff
@@ -11,10 +11,15 @@ void OnigString::Init(Local<Object> target) {
 }
 
 NAN_METHOD(OnigString::New) {
-  Nan::HandleScope scope;
-  OnigString* scanner = new OnigString(Local<String>::Cast(info[0]));
-  scanner->Wrap(info.This());
-  info.GetReturnValue().SetUndefined();
+  Local<String> content = Local<String>::Cast(info[0]);
+  if (!content->IsString()) {
+    Nan::ThrowTypeError("Argument must be a string");
+    return;
+  }
+
+  OnigString* string = new OnigString(content);
```

Comment:
Looks like this might be leaking, unless there's some magic going on in `node::ObjectWrap`? :potable_water:
Regardless, it can be modernized to using smart pointers to guard against leaks;
``` c++
std::shared_ptr<OnigString> string = std::make_shared<OnigString>(content);
```

Personally I also would have used the `auto` keyword here, but we don't seem to be using that in any of our C++ projects, hence the full type. And note that if the `Wrap` function does perform magic, smart pointers might not work :8ball: